build-system/tasks/runtime-test/helpers-test-reporting.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const fetch = require('node-fetch');
+const fs = require('fs');
+const log = require('fancy-log');
+const JSON_REPORT_TEST_TYPES = new Set(['unit', 'integration']);
+const REPORTING_API_URL = 'https://amp-test-cases.uc.r.appspot.com/report';
+const {
+  travisBuildNumber,
+  travisJobNumber,
+  travisCommitSha,
+} = require('../../common/travis');
+
+function resultFilename(testType) {
+  return `results_${testType}.json`;
+}
+
+async function getReports() {
+  const reportPromises = JSON_REPORT_TEST_TYPES.map(async (testType) => {
+    const jsonString = await fs.readFile(resultFilename(testType));
+    const reportJson = JSON.parse(jsonString);
+    return {testType, reportJson};
+  });
+
+  const incompleteReports = await Promise.all(reportPromises);
+  const reports = incompleteReports.map(addJobAndBuildInfo);
+
+  return reports;
+}
+
+function addJobAndBuildInfo({testType, reportJson}) {
+  const build = {
+    buildNumber: travisBuildNumber(),
+    commitHash: travisCommitSha(),
+    startedAt: 0,
```

Comment:
Think about whether we _do_ need startedAt for builds & jobs and whether this could be computed in the app's end. (If the latter, should it be a nullable field in the Build & Job types?)

Reply:
Is this information available in the Karma test report data?

Reply:
the startedAt for the build is not available in the Karma test report data. It could be obtained by making a call to the Travis API, but I think it would be better to have that in the app, to minimize the amount of logic on this end.

Reply:
For now I will only send the build number and commit hash; if the application wants more information, it can compute it from those.